apps/documents/views.py

```diff
@@ -237,25 +237,75 @@ def get_success_url(self):
+                    messages.success(
+                        self.request, "Collection indexing has been enabled. Files will be indexed in the background."
+                    )
 
-        collection = form.instance
-        old_vector_store_id = collection.openai_vector_store_id
+            response = super().form_valid(form)
 
-        if form.instance.is_index and form.instance.is_remote_index and "llm_provider" in form.changed_data:
-            with transaction.atomic():
+            # Handle LLM provider changes for existing indexed collections
+            if form.instance.is_index and form.instance.is_remote_index and "llm_provider" in form.changed_data:
                 collection.openai_vector_store_id = None  # Reset the vector store ID
                 collection.ensure_remote_index_created()
                 CollectionFile.objects.filter(collection_id=collection.id).update(status=FileStatus.PENDING)
 
-            tasks.migrate_vector_stores.delay(
-                collection_id=form.instance.id,
-                from_vector_store_id=old_vector_store_id,
-                from_llm_provider_id=form.initial["llm_provider"],
-            )
+                tasks.migrate_vector_stores.delay(
+                    collection_id=form.instance.id,
+                    from_vector_store_id=old_vector_store_id,
+                    from_llm_provider_id=form.initial["llm_provider"],
+                )
 
         return response
 
+    def _delete_indexed_data(self, collection):
+        """Delete all indexed data for a collection"""
+        if collection.is_remote_index and collection.openai_vector_store_id:
+            # Delete remote index
+            index_manager = collection.get_index_manager()
+            index_manager.delete_remote_index()
+            collection.openai_vector_store_id = ""
+            collection.save(update_fields=["openai_vector_store_id"])
+        else:
+            # Delete local index data
+            from apps.files.models import FileChunkEmbedding
+
+            FileChunkEmbedding.objects.filter(collection_id=collection.id).delete()
+
+        # Reset collection file statuses
+        CollectionFile.objects.filter(collection_id=collection.id).update(status="")
+
+    def _setup_indexing(self, collection, form):
+        """Setup indexing for a collection that's being converted to indexed"""
+        if form.cleaned_data.get("is_remote_index", False):
+            collection.ensure_remote_index_created()
+
+        # Set up collection files for indexing
+        collection_files = CollectionFile.objects.filter(collection_id=collection.id)
+        if collection_files.exists():
+            # Update metadata and status for existing files
+            chunking_strategy = ChunkingStrategy(chunk_size=800, chunk_overlap=400)
+            metadata = CollectionFileMetadata(chunking_strategy=chunking_strategy)
+
+            collection_files.update(status=FileStatus.PENDING, metadata=metadata)
```

Comment:
There might be a mismatch between the set of file types in the non-indexed collection and the index's supported file types. I'm thinking we should filter for only supported file types and only try to index those. We should tell the user about this in the notice we show them when they want to do the switch.

Q: What should happen with the rest?
A: We can either remove them, or leave them with an empty status. Or add a new "unsupported" status. Maybe this makes more sense, because then the user can hover over these files and see why they cannot be indexed.